src/types.ts

```diff
@@ -161,17 +161,31 @@ export async function iterativelyCheckStatus(
     txHash: string,
     starknetWrapper: StarknetWrapper,
     resolve: (status: string) => void,
-    reject: (reason: Error) => void
+    reject: (reason: Error) => void,
+    retryCount = 10
 ) {
-    const statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
-        warn(reason);
-        return undefined;
-    });
+    let statusObject;
+    let errorReason;
+    let count = retryCount;
+
+    while (count > 0) {
+        statusObject = await checkStatus(txHash, starknetWrapper).catch((reason) => {
```

Comment:
A comment could be added before this line that this promise is rejected usually if the network is unavailable.